### PR TITLE
Create Longest_Substring_Without_Repeating_Characters.cpp

### PR DESCRIPTION
**Pull request by RohithaAiswarya16**
Added a new solution file Longest_Substring_Without_Repeating_Characters.cpp, which implements a C++ program to find the length of the longest substring without repeating characters in a given string. This program uses a sliding window approach along with a hash map to store the last seen index of each character, allowing for efficient tracking of unique substrings.

The code takes user input for the string and outputs the maximum length of any substring without duplicate characters. Comments and an example are included for clarity.

### CODE REVIEW

File: Solved-Problems/Longest_Substring_Without_Repeating_Characters/Longest_Substring_Without_Repeating_Characters.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - Add "c" -> "abc" -> length 3
+    - "a" repeats, so move the start to the next character -> "bca" (length still 3)
+    - Continue until the end, with the longest substring remaining "abc".
+Output: 3
+
+Input: "bbbbb"
+Explanation:
+- The longest substring without repeating characters is "b", which has a length of 1.
+Output: 1
+*/
```
Comment (iamwatchdogs) [inline Solved-Problems/Longest_Substring_Without_Repeating_Characters/Longest_Substring_Without_Repeating_Characters.cpp:57]: Hi @RohithaAiswarya16, it seems like some of the linting checks have failed due to your unformatted source code. So, Please make the following changes to pass the linting checks...

<details>
<summary><h2>View Changes</h2></summary>

> Don't worry, It's still your code but formatted one...

```suggestion
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

// Function to find the length of the longest substring without repeating
// characters
int lengthOfLongestSubstring(const string &s) {
  unordered_map<char, int> charIndexMap;
  int maxLength = 0;
  int start = 0;

  for (int end = 0; end < s.length(); end++) {
    char currentChar = s[end];

    // If the character is already in the map and is within the current window
    if (charIndexMap.find(currentChar) != charIndexMap.end() &&
        charIndexMap[currentChar] >= start) {
      start = charIndexMap[currentChar] + 1;
    }

    // Update the last seen index of the current character
    charIndexMap[currentChar] = end;

    // Calculate the max length of substring
    maxLength = max(maxLength, end - start + 1);
  }
  return maxLength;
}

int main() {
  string input;
  cout << "Enter a string: ";
  cin >> input;

  int result = lengthOfLongestSubstring(input);
  cout
      << "The length of the longest substring without repeating characters is: "
      << result << endl;

  return 0;
}

/*
Example:
Input: "abcabcbb"
Explanation:
- The longest substring without repeating characters is "abc", which has a
length of 3.
- As we iterate through the string:
    - Start with "a" -> length 1
    - Add "b" -> "ab" -> length 2
    - Add "c" -> "abc" -> length 3
    - "a" repeats, so move the start to the next character -> "bca" (length
still 3)
    - Continue until the end, with the longest substring remaining "abc".
Output: 3

Input: "bbbbb"
Explanation:
- The longest substring without repeating characters is "b", which has a length
of 1. Output: 1
*/
```

</details>

**Review by iamwatchdogs — CHANGES_REQUESTED** [review]
Please make the required changes so that linting checks would pass.

**Comment by RohithaAiswarya16** [comment]
Thank you for the updated code! I’ve applied all the suggested changes and pushed the updates. Please review and let me know if further adjustments are needed.

**Review by iamwatchdogs — APPROVED** [review]
Reviewed and Approved by @iamwatchdogs.

**Pull request merged**